stream.go

```diff
@@ -554,8 +568,9 @@ func bulkAppendFields(w io.Writer, ws *xlsxWorksheet, from, to int) {
 // is written to the temp file with Sync, which may return an error.
 // Therefore, Sync should be periodically called and the error checked.
 type bufferedWriter struct {
-	tmp *os.File
-	buf bytes.Buffer
+	tmp    *os.File
+	buf    bytes.Buffer
+	header bytes.Buffer
```

Comment:
this was the best way I found to allow changing things on the xml that comes before the data
I tought about reserving a bit of space, 4 or 8 kb, at the begining of the buffer, but this could cause problems, the amount of data will depend on the user after all?
is `prefix` a better name? along with a explnation on why 2 buffers are needed?

Reply:
Allocation an new buffer will bring maintenance costs, and it is uncertain whether functions such as sheet views, sheet format properties, columns settings, page margins, header, and footers will be maintained in it in the future. Using one buffer makes it easy to track and maintain the consistency of the contextual content of the data to be streamed.

Reply:
It increases the maintenance cost but it also allows more freedom to modify the worksheet object
I only did the 2 buffer thing to maintain the ordering of the xml elements and avoid having to copy the sheet data just to prepend a few KBs, as that would be costly, the buffer with the first piece of the XML with the elements that come before the sheet data (the badly named `header`) is only writen on the Flush method
As long as we only use the exported methods of the `bufferedWriter` (accessing the buffer directly already caused problems when the temp file was being used) and don't try to read the sheet data before flushing (this was already a no-no before the changes) we should be fine. 
If you want to I can do a detailed breakdown on why I did it this way